### PR TITLE
chore: 🤖 refactor

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
1. Add Copy trait on `Syntax` , so we don't need to call `.clone()` 
2. bump criterion
3. enable `LTO` in release mode

### CODE REVIEW

**Comment by g-plane** [comment]
LTO option has no effect on downstream users, so we don't need it.

**Comment by IWANABETHATGUY** [comment]
> 

Maybe putting it in `bench` category is fine.

File: raffia/src/parser/builder.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ use crate::{
 /// Parser builder is for building a parser while allowing us
 /// to control advanced behaviors.
 ///
-/// Unlike [`Parser`], syntax isn't required when creating a parser builder,
+/// Unlike [Parser], syntax isn't required when creating a parser builder,
```
Comment (g-plane) [inline raffia/src/parser/builder.rs:10]: Please don't remove the link.

File: raffia/src/config.rs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 use serde::{Deserialize, Serialize};
 
 /// Supported syntax.
-#[derive(Clone, Debug, Default, PartialEq, Eq)]
+#[derive(Clone, Debug, Default, PartialEq, Eq, Copy)]
```
Comment (g-plane) [inline raffia/src/config.rs:5]:
```suggestion
#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
```

**Review by g-plane — CHANGES_REQUESTED** [review]

File: raffia/src/parser/builder.rs (hunk 1 above)
Comment (IWANABETHATGUY) [reply raffia/src/parser/builder.rs]: image
code block not necessary if you only want the link,

Comment (IWANABETHATGUY) [reply raffia/src/parser/builder.rs]: But I have reverted it.

**Comment by g-plane** [comment]
Have you run benchmark? I'm curious about your benchmark result at your computer. (Don't forget to paste your computer spec as well.)

**Comment by IWANABETHATGUY** [comment]
I am not sure it is meaningful to paste it here, because we have different bench suites.

```
Gnuplot not found, using plotters backend
parse_selector/Raffia/*|a-bc--xx#q09h6nn.--oq323r23r > |x76j5 + [mmmmm][oew=a i][\0065='lll'], u6\z:...                                                                             
                        time:   [1.8585 µs 1.8752 µs 1.8922 µs]
                        change: [-4.5692% -4.0649% -3.5503%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 17 outliers among 100 measurements (17.00%)
  14 (14.00%) low mild
  3 (3.00%) high mild
parse_selector/LightningCSS/*|a-bc--xx#q09h6nn.--oq323r23r > |x76j5 + [mmmmm][oew=a i][\0065='lll'],...                                                                             
                        time:   [2.0598 µs 2.0702 µs 2.0794 µs]
                        change: [-5.5851% -5.1689% -4.7740%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild
parse_selector/SWC/*|a-bc--xx#q09h6nn.--oq323r23r > |x76j5 + [mmmmm][oew=a i][\0065='lll'], u6\z:hov...                                                                             
                        time:   [3.1286 µs 3.1330 µs 3.1382 µs]
                        change: [+1.9655% +2.1302% +2.3127%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 5 outliers among 100 measurements (5.00%)
  5 (5.00%) high mild

     Running benches/parse_stylesheet.rs (target/release/deps/parse_stylesheet-bbbdee7737b2cd32)
WARNING: HTML report generation will become a non-default optional feature in Criterion.rs 0.4.0.
This feature is being moved to cargo-criterion (https://github.com/bheisler/cargo-criterion) and will be optional in a future version of Criterion.rs. To silence this warning, either switch to cargo-criterion or enable the 'html_reports' feature in your Cargo.toml.

Gnuplot not found, using plotters backend
parser_stylesheet/Raffia/bootstrap-utilities.css                                                                            
                        time:   [1.1044 ms 1.1064 ms 1.1088 ms]
                        change: [-10.003% -9.6971% -9.4078%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 10 outliers among 100 measurements (10.00%)
  8 (8.00%) high mild
  2 (2.00%) high severe
parser_stylesheet/LightningCSS/bootstrap-utilities.css                                                                            
                        time:   [1.1878 ms 1.1942 ms 1.1990 ms]
                        change: [-0.5459% -0.3402% -0.1612%] (p = 0.00 < 0.05)
                        Change within noise threshold.
Found 6 outliers among 100 measurements (6.00%)
  2 (2.00%) low severe
  4 (4.00%) high mild
parser_stylesheet/SWC/bootstrap-utilities.css                                                                            
                        time:   [1.7723 ms 1.7799 ms 1.7878 ms]
                        change: [-9.2115% -8.7085% -8.1761%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 2 outliers among 100 measurements (2.00%)
  1 (1.00%) high mild
  1 (1.00%) high severe
parser_stylesheet/Raffia/bootstrap-reboot.css                                                                             
                        time:   [112.33 µs 112.70 µs 113.10 µs]
                        change: [+4.5336% +4.8749% +5.2123%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 3 outliers among 100 measurements (3.00%)
  2 (2.00%) high mild
  1 (1.00%) high severe
parser_stylesheet/LightningCSS/bootstrap-reboot.css                                                                            
                        time:   [179.88 µs 180.48 µs 181.25 µs]
                        change: [-4.5597% -3.2608% -2.2559%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 7 outliers among 100 measurements (7.00%)
  4 (4.00%) high mild
  3 (3.00%) high severe
parser_stylesheet/SWC/bootstrap-reboot.css                                                                            
                        time:   [218.79 µs 219.11 µs 219.46 µs]
                        change: [-6.2901% -5.9303% -5.5962%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 5 outliers among 100 measurements (5.00%)
  4 (4.00%) high mild
  1 (1.00%) high severe
parser_stylesheet/Raffia/tailwind.css                                                                            
                        time:   [76.630 ms 77.200 ms 77.807 ms]
                        change: [-0.5750% +0.3144% +1.2774%] (p = 0.51 > 0.05)
                        No change in performance detected.
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high severe
parser_stylesheet/LightningCSS/tailwind.css                                                                            
                        time:   [89.540 ms 90.277 ms 90.999 ms]
                        change: [-3.0918% -2.1537% -1.2398%] (p = 0.00 < 0.05)
                        Performance has improved.
Benchmarking parser_stylesheet/SWC/tailwind.css: Warming up for 3.0000 s
Warning: Unable to complete 100 samples in 12.0s. You may wish to increase target time to 12.6s, or reduce sample count to 90.
parser_stylesheet/SWC/tailwind.css                                                                            
                        time:   [125.19 ms 125.72 ms 126.30 ms]
                        change: [-8.3288% -7.9397% -7.5198%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 11 outliers among 100 measurements (11.00%)
  5 (5.00%) high mild
  6 (6.00%) high severe

     Running unittests src/lib.rs (target/release/deps/raffia_macro-ae2276e08e805cad)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running unittests src/lib.rs (target/release/deps/wasm-b9513bb25187fa9b)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

Linux 
amd 5950x

**Review by g-plane — APPROVED** [review]

**Pull request merged**